### ISSUE TITLE
[Bug] Fail pod when change numbers of podes

### ISSUE THREAD

**Issue by tsyrfa** [strimzi/strimzi-kafka-operator#2494]
**Describe the bug**
When I change numbers of replica of kafka infrastructure one of pods can't to start after recreating (it always kafka-cluster-kafka-1)

**To Reproduce**

1. Create a separate namespace:
`kubectl create namespace kafka-test`

2. Install the cluster-operator (I pull the "cluster-operator" folder from your repo)
`kubectl apply -f .\cluster-operator`

3. Apply configure for kafka-cluster

```
apiVersion: kafka.strimzi.io/v1beta1
 kind: Kafka
 metadata:
  name: kafka-cluster
 spec:
   kafka:
     replicas: 3
     listeners:
      plain: {}
      tls: {}
      external:
        type: nodeport
        tls: false

          
    storage:
      type: ephemeral
      
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1 
    
  zookeeper:
    replicas: 3
        
    storage:
      type: ephemeral
      
    entityOperator:
      topicOperator: {}
      userOperator: {}

```
4. Wait while it finished

5. Change numbers of replica for kafka and zookeeper:

```...
spec:
  kafka:
    replicas: 4
...
  zookeeper:
    replicas: 4
```

full configure is here:

```
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: kafka-cluster
spec:
  kafka:
    replicas: 4
    listeners:
      plain: {}
      tls: {}
      external:
        type: nodeport
        tls: false

          
    storage:
      type: ephemeral
      
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1 
    
  zookeeper:
    replicas: 4
        
    storage:
      type: ephemeral
      
    entityOperator:
      topicOperator: {}
      userOperator: {}
```

6. `kafka-cluster-kafka-1` return error (it always number pod 1)

**Expected behavior**
All pods correct started, when I changed numbers of replica

**Environment (please complete the following information):**
 - Strimzi version: 0.16.0 and 0.16.2 
 - Installation method: YAML files
 - Kubernetes cluster:  Kubernetes 1.15
 - Infrastructure: "yandex cloud"

**YAML files and logs**

This is output after change numbers of replic
[get pod.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143116/get.pod.txt)

Logs from crashed Kafka:
[kafka-cluster-kafka-1.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143118/kafka-cluster-kafka-1.txt)

Logs from corrected started Kafka:
[kafka-cluster-kafka-0.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143119/kafka-cluster-kafka-0.txt)

Logs from crashed Kafka tls-sidecar:
[kafka-cluster-kafka-1_tls-sidecar.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143124/kafka-cluster-kafka-1_tls-sidecar.txt)

Logs from correct started Kafka tls-sidecar:
[kafka-cluster-kafka-0_tls-sidecar.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143123/kafka-cluster-kafka-0_tls-sidecar.txt)

Logs from Zookeeper :
[kafka-cluster-zookeeper-0.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143127/kafka-cluster-zookeeper-0.txt)

Logs from Zookeeper tls-sidecar :
[kafka-cluster-zookeeper-0-tls-sidecar.txt](https://github.com/strimzi/strimzi-kafka-operator/files/4143128/kafka-cluster-zookeeper-0-tls-sidecar.txt)

**Additional context**
At first glance it looks like same bug:
https://github.com/strimzi/strimzi-kafka-operator/issues/2449

**Comment by scholzj** [comment]
This is weird especially if only one pods suffers from it. Do you have anything else running on the same node as the crashing pod? Can it connect to the outside?

Also, you are using epheemral storage which gets basically lost everytime you restart the pod. You should really so something like this with persistent storage. With the ephemeral lot of things can go wrong when you roll the pods.

**Comment by tsyrfa** [comment]
Yes, I have on this node few different pods.
What interesting, in same node placed other replicas, that works

```
NAMESPACE        NAME                                        READY   STATUS             RESTARTS   AGE   IP              NODE                        NOMINATED NODE   READINESS GATES
kafka-test       kafka-cluster-kafka-0                       2/2     Running            0          87m   10.155.134.41   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-test       kafka-cluster-kafka-1                       1/2     CrashLoopBackOff   20         82m   10.155.134.46   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-test       kafka-cluster-kafka-2                       2/2     Running            0          85m   10.155.134.43   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-test       kafka-cluster-zookeeper-0                   2/2     Running            0          84m   10.155.134.44   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-test       kafka-cluster-zookeeper-1                   2/2     Running            0          83m   10.155.134.45   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
```

I'll try to convert it to `persistent storage`, testing it and let you know

**Comment by scholzj** [comment]
The issue above mentions 4 Zookeeper nodes. But this listing does have only 2 Zookeeper pods. Is that just that you copied only part of the output? Or are the other two ZK pods really missing? Because 2 Zookeeper pods out of 4 would not be quorum and it would not work properly.

**Comment by tsyrfa** [comment]
I show all pods in same namespace on node `cl1jomaq0a7tc152o3ph-ypys`
Missing only `kafka-cluster-kafka-1` (`kafka-cluster-kafka-2` can't updating and `kafka-cluster-kafka-3` cant creating becouse they wait while `kafka-cluster-kafka-1` will become in `running` state)

So, now I created 4 kafka-servers and 4 zookeeper pods with persistent-claim. changed their number to 5, and I saw the same problem:

```
c:\k8s\kafka>kubectl get pod -n kafka-test -o wide
NAME                                        READY   STATUS             RESTARTS   AGE     IP              NODE                        NOMINATED NODE   READINESS GATES
kafka-cluster-kafka-0                       2/2     Running            0          15m     10.155.134.49   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     CrashLoopBackOff   5          5m48s   10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-2                       2/2     Running            0          15m     10.155.134.50   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-3                       2/2     Running            0          15m     10.155.133.39   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-0                   2/2     Running            0          10m     10.155.134.52   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-1                   2/2     Running            0          10m     10.155.132.30   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-2                   2/2     Running            0          8m28s   10.155.133.41   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-3                   2/2     Running            0          6m49s   10.155.134.53   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-4                   2/2     Running            0          11m     10.155.133.40   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
strimzi-cluster-operator-6975d8874f-ccd4d   1/1     Running            0          22h     10.155.134.33   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
```
It's a order of updating:

```
c:\k8s\kafka>kubectl get pod -n kafka-test -o wide -w
NAME                                        READY   STATUS    RESTARTS   AGE     IP              NODE                        NOMINATED NODE   READINESS GATES
kafka-cluster-kafka-0                       2/2     Running   0          4m41s   10.155.134.49   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       2/2     Running   0          4m41s   10.155.134.51   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-2                       2/2     Running   0          4m41s   10.155.134.50   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-3                       2/2     Running   0          4m41s   10.155.133.39   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-0                   2/2     Running   0          5m34s   10.155.134.48   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-1                   2/2     Running   0          5m34s   10.155.132.29   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-2                   2/2     Running   0          5m34s   10.155.133.38   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-3                   2/2     Running   0          5m34s   10.155.134.47   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-4                   0/2     Running   0          42s     10.155.133.40   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
strimzi-cluster-operator-6975d8874f-ccd4d   1/1     Running   0          22h     10.155.134.33   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-4                   1/2     Running   0          48s     10.155.133.40   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-4                   2/2     Running   0          49s     10.155.133.40   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-0                   2/2     Terminating   0          5m42s   10.155.134.48   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   0/2     Terminating   0          5m43s   10.155.134.48   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   0/2     Terminating   0          5m47s   10.155.134.48   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   0/2     Terminating   0          5m47s   10.155.134.48   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   0/2     Pending       0          0s      <none>          <none>                      <none>           <none>
kafka-cluster-zookeeper-0                   0/2     Pending       0          0s      <none>          cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   0/2     ContainerCreating   0          0s      <none>          cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   0/2     Running             0          11s     10.155.134.52   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   1/2     Running             0          27s     10.155.134.52   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-0                   2/2     Running             0          27s     10.155.134.52   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-1                   2/2     Terminating         0          6m15s   10.155.132.29   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Terminating         0          6m17s   10.155.132.29   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Terminating         0          6m17s   10.155.132.29   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Terminating         0          6m25s   10.155.132.29   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Terminating         0          6m25s   10.155.132.29   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Pending             0          0s      <none>          <none>                      <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Pending             0          0s      <none>          cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     ContainerCreating   0          0s      <none>          cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   0/2     Running             0          71s     10.155.132.30   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   1/2     Running             0          86s     10.155.132.30   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-1                   2/2     Running             0          93s     10.155.132.30   cl1jomaq0a7tc152o3ph-ulyp   <none>           <none>
kafka-cluster-zookeeper-2                   2/2     Terminating         0          7m58s   10.155.133.38   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Terminating         0          8m      10.155.133.38   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Terminating         0          8m      10.155.133.38   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Terminating         0          8m7s    10.155.133.38   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Terminating         0          8m7s    10.155.133.38   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Pending             0          0s      <none>          <none>                      <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Pending             0          0s      <none>          cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     ContainerCreating   0          0s      <none>          cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   0/2     Running             0          75s     10.155.133.41   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   1/2     Running             0          94s     10.155.133.41   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-2                   2/2     Running             0          95s     10.155.133.41   cl1jomaq0a7tc152o3ph-okag   <none>           <none>
kafka-cluster-zookeeper-3                   2/2     Terminating         0          9m43s   10.155.134.47   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Terminating         0          9m45s   10.155.134.47   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Terminating         0          9m45s   10.155.134.47   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Terminating         0          9m46s   10.155.134.47   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Terminating         0          9m46s   10.155.134.47   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Pending             0          0s      <none>          <none>                      <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Pending             0          0s      <none>          cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     ContainerCreating   0          0s      <none>          cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   0/2     Running             0          18s     10.155.134.53   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   1/2     Running             0          33s     10.155.134.53   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-zookeeper-3                   2/2     Running             0          38s     10.155.134.53   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       2/2     Terminating         0          9m34s   10.155.134.51   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Terminating         0          9m43s   10.155.134.51   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Terminating         0          9m43s   10.155.134.51   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Terminating         0          9m54s   10.155.134.51   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Terminating         0          9m54s   10.155.134.51   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Pending             0          0s      <none>          <none>                      <none>           <none>
kafka-cluster-kafka-1                       0/2     Pending             0          0s      <none>          cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Init:0/1            0          0s      <none>          cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Init:0/1            0          27s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     PodInitializing     0          29s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Running             0          30s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Error               0          40s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Running             1          41s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       0/2     Error               1          50s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Error               1          51s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     CrashLoopBackOff    1          54s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Running             2          67s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Error               2          75s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     CrashLoopBackOff    2          84s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Running             3          96s     10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Error               3          105s    10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     CrashLoopBackOff    3          114s    10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Running             4          2m35s   10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     Error               4          2m43s   10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
kafka-cluster-kafka-1                       1/2     CrashLoopBackOff    4          2m44s   10.155.134.54   cl1jomaq0a7tc152o3ph-ypys   <none>           <none>
```

**Comment by tsyrfa** [comment]
On same node I have kafka pods in other namespace. But I think that it cant to affect on my cluster, because I created namespace "kafka-test" after discovered problem with scaling for test it

**Comment by tsyrfa** [comment]
So if I reduce the number of pods, everything will work correctly.
If I increase the number of only kafka or only the zoo, everything works correctly
If I increase them together, one of kafka (now it is kafka-0) will return an error
When a configuration change causes a rolling update, this is also true. (for example, if I adding block `limits`)

Maybe this is a workaround for me.

**Comment by scholzj** [comment]
Hmm, ok, I can acknowledge this. The Zookeeper scaleup does not seem to work even when you do it on its own - and that seems to be with both Kafka 2.3.1 and 2.4.0. I'm not sure exactly what the problem is.

@tombentley @tomncooper Could you look at this and seee if you have any ideas?

**Comment by scholzj** [comment]
Fixed in #2574. Thanks for reporting this @tsyrfa
